File: data/python/MAXIMUM_NUMBER_CHOCOLATES_DISTRIBUTED_EQUALLY_AMONG_K_STUDENTS_TRANSLATED.java
====================
public static int f_gold ( int [ ] arr , int n , int k ) {
  int [ ] um = new int [ n ] , currRem = 0 , maxSum = 0 ;
  int [ ] sm = new int [ n ] ;
  sm [ 0 ] = arr [ 0 ] ;
  for ( int i = 1 ;
  i < n ;
  i ++ ) sm [ i ] = sm [ i - 1 ] + arr [ i ] ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    currRem = sm [ i ] % k ;
    if ( ( ! currRem && maxSum < sm [ i ] ) || ( ! currRem && maxSum < um [ currRem ] ) ) maxSum = sm [ i ] ;
    else if ( ( ! currRem && maxSum < um [ currRem ] ) || ( ! currRem && maxSum < um [ currRem ] ) ) um [ currRem ] = i ;
    else if ( ( maxSum < ( sm [ i ] - sm [ um [ currRem ] ] ) ) || ( ! currRem && maxSum < um [ currRem ] ) ) maxSum = sm [ i ] - sm [ um [ currRem ] ] ;
  }
  return maxSum / k ;
}

File: data/java/MAXIMUM_NUMBER_CHOCOLATES_DISTRIBUTED_EQUALLY_AMONG_K_STUDENTS_TRANSLATED.py
====================
def f_gold ( arr , n , k ) :
    um = { }
    sum = [ ]
    curr_rem = 0
    max_sum = 0
    sum.append ( arr [ 0 ] )
    for i in range ( 1 , n ) :
        sum [ i ] = sum [ i - 1 ] + arr [ i ]
    for i in range ( 0 , n ) :
        curr_rem = sum [ i ] % k
        if curr_rem == 0 :
            if max_sum < sum [ i ] :
                max_sum = sum [ i ]
        elif not um.has_key ( curr_rem ) :
            um [ curr_rem ] = i
        elif max_sum < ( sum [ i ] - sum [ um [ curr_rem ] ] ) :
            max_sum = sum [ i ] - sum [ um [ curr_rem ] ]
    return ( max_sum / k )

